src/doppler/sinks/syslogwriter/https_writer_test.go

```diff
@@ -83,7 +91,76 @@ var _ = Describe("HttpsWriter", func() {
 			parsedTime, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
 			_, err = w.Write(standardErrorPriority, []byte("Message"), "just a test", "TEST", parsedTime.UnixNano())
 			Expect(err).ToNot(HaveOccurred())
+		})
 
+		Describe("idle connection behavior", func() {
+			var (
+				receivedChan chan []byte
+				waitChan     chan struct{}
+				listener     *historyListener
+			)
+
+			BeforeEach(func() {
+				receivedChan = make(chan []byte, 1)
+				waitChan = make(chan struct{})
+
+				l, err := net.Listen("tcp", "127.0.0.1:0")
+				Expect(err).NotTo(HaveOccurred())
+
+				listener = &historyListener{Listener: l}
+				server.Listener = listener
+
+				slowHandler := func(_ http.ResponseWriter, r *http.Request) {
```

Comment:
Using these three channels (`waitChan`, `receivedChan`, and `completedReqs`) is complicated. It doesn't help that it all depends on the fact that `receivedChan` is a buffered channel with size, which results in blocking. We also never read any data from any of these channels (they're all just for signals).

Could the handler function be turned into an object, that has `Pause()` and `Finish()` functions? If done properly, all these channels, if still necessary, would be internal the object and not used in the test.